flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/TableEnvImpl.scala

```diff
@@ -69,45 +66,19 @@ import _root_.scala.collection.mutable
   */
 abstract class TableEnvImpl(val config: TableConfig) extends TableEnvironment {
 
-  // the catalog to hold all registered and translated tables
-  // we disable caching here to prevent side effects
-  private val internalSchema: CalciteSchema = CalciteSchema.createRootSchema(false, false)
-  private val rootSchema: SchemaPlus = internalSchema.plus()
-
   // Table API/SQL function catalog
   private[flink] val functionCatalog: FunctionCatalog = new FunctionCatalog()
 
-  // the configuration to create a Calcite planner
-  private lazy val frameworkConfig: FrameworkConfig = Frameworks
-    .newConfigBuilder
-    .defaultSchema(rootSchema)
-    .parserConfig(getSqlParserConfig)
-    .costFactory(new DataSetCostFactory)
-    .typeSystem(new FlinkTypeSystem)
-    .operatorTable(getSqlOperatorTable)
-    .sqlToRelConverterConfig(getSqlToRelConverterConfig)
-    // the converter is needed when calling temporal table functions from SQL, because
-    // they reference a history table represented with a tree of table operations
-    .context(Contexts.of(
-      new TableOperationConverter.ToRelConverterSupplier(expressionBridge)
-    ))
-    // set the executor to evaluate constant expressions
-    .executor(new ExpressionReducer(config))
-    .build
+  private val BUILTIN_CATALOG_NAME = "builtin"
```

Comment:
does TableEnvironment need to know the default catalog's name in CatalogManager?

Reply:
Does `CatalogManager` needs to know about it? This is just a initial value for the current catalog, so I think it fits better to the `TableEnvironment` rather than `CatalogManager`.